plugins/optimization-detective/storage/data.php

```diff
@@ -143,19 +143,55 @@ function od_get_url_metrics_slug( array $query_vars ): string {
 /**
  * Gets the current ETag for URL Metrics.
  *
- * The ETag is a hash based on the IDs of the registered tag visitors
- * in the current environment. It is used for marking the URL Metrics as stale
- * when its value changes.
+ * Generates a hash based on the IDs of registered tag visitors, the queried object,
+ * posts in The Loop, and theme information in the current environment. This ETag
+ * is used to assess if the URL Metrics are stale when its value changes.
  *
  * @since n.e.x.t
  * @access private
  *
- * @param OD_Tag_Visitor_Registry $tag_visitor_registry Tag visitor registry.
+ * @param OD_Tag_Visitor_Registry       $tag_visitor_registry Tag visitor registry.
+ * @param WP_Query                      $wp_query             The WP_Query instance.
+ * @param string|WP_Block_Template|null $current_template     The current template being used.
  * @return non-empty-string Current ETag.
  */
-function od_get_current_url_metrics_etag( OD_Tag_Visitor_Registry $tag_visitor_registry ): string {
+function od_get_current_url_metrics_etag( OD_Tag_Visitor_Registry $tag_visitor_registry, WP_Query $wp_query, $current_template ): string {
+	$queried_object      = $wp_query->get_queried_object();
+	$queried_object_data = array(
+		'id'            => null,
+		'type'          => null,
+		'last_modified' => null,
+	);
+
+	if ( $queried_object instanceof WP_Post ) {
+		$queried_object_data['id']            = $queried_object->ID;
+		$queried_object_data['type']          = 'post';
+		$queried_object_data['last_modified'] = $queried_object->post_modified_gmt;
+	} elseif ( $queried_object instanceof WP_Term ) {
+		$queried_object_data['id']   = $queried_object->term_id;
+		$queried_object_data['type'] = 'term';
+	} elseif ( $queried_object instanceof WP_User ) {
+		$queried_object_data['id']   = $queried_object->ID;
+		$queried_object_data['type'] = 'user';
+	} elseif ( $wp_query->is_post_type_archive() ) {
+		$queried_object_data['type'] = $wp_query->get( 'post_type' );
```

Comment:
What about:

```suggestion
	} elseif ( $queried_object instanceof WP_Post_Type ) {
		$queried_object_data['type'] = $queried_object->name;
```

Reply:
This looks good to me. I realize this is how I originally wrote it—I must have changed it at some point but can't recall why.